### ISSUE TITLE
.NET Framework Requirement

### ISSUE THREAD

**Issue by RadWolfie** [Cxbx-Reloaded/Cxbx-Reloaded#1710]
I'm creating this issue open for further investigation to find out how come some contributors are having issues to compile C# projects, and maybe CMake's C# project generations.

Since the main emulator's relative projects does not use C#. Except for Cxbx-Debugger's relative projects does. It may be possible to exclude C# projects if C# compiler is not detected within CMakeList. Or at least fix the C# requirement.

There's a chance the issue may be coming from submodule's repo. If it does, then will need to create a ticket there and reference to this ticket.

---

At of this time of issue created, I do not remember the whole details. Since I only came across this issue couple times in the past when CMakeList files were created. Which usually caused by Visual Studio installer's update.

**Comment by x1nixmzeng** [comment]
The debugger has a dependency on cs_x86, which is a submodule for capstone disassembly

**Comment by x1nixmzeng** [comment]
The debugger project (and its deps) should made an optional cmake build option to avoid build environments without C# support

**Comment by RadWolfie** [comment]
True, except CMake is currently using a hack for Visual Studio environment and doesn't have Mono software support. Once CMake has better support for detecting C# compiler, then we will implement better support to be aware if C# compiler is installed or not.

Luke had implemented exactly the same way what CMake did internally. See below:
https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/blob/0c470044fbfbf07c9d04bae706760672e8ea4f16/CMakeLists.txt#L367

Although, we could add an option to "force" not to generate C# project files in CMake's CLI.
